dask_ml/preprocessing/label.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import division
+
+from operator import getitem
+
+import dask.array as da
+import dask.dataframe as dd
+import numpy as np
+from sklearn.preprocessing import label as sklabel
+from sklearn.utils.validation import check_is_fitted
+
+
+class LabelEncoder(sklabel.LabelEncoder):
+
+    __doc__ = sklabel.LabelEncoder.__doc__
+
+    def fit(self, y):
+        if isinstance(y, dd.Series):
+            y = da.asarray(y)
+
+        if isinstance(y, da.Array):
+            classes_ = da.unique(y)
+            classes_ = classes_.compute()
+        else:
+            classes_ = np.unique(y)
+
+        self.classes_ = classes_
+
+        return self
+
+    def fit_transform(self, y):
+        return self.fit(y).transform(y)
+
+    def transform(self, y):
+        check_is_fitted(self, 'classes_')
+        if isinstance(y, dd.Series):
+            y = da.asarray(y)
+
+        if isinstance(y, da.Array):
+            return y.map_blocks(lambda b: np.searchsorted(self.classes_, b),
+                                dtype=self.classes_.dtype)
```

Comment:
Can I suggest placing `self.classes_` as a keyword argument and dropping the lambda?  Maybe something like the following:

```python
return y.map_blocks(lambda block, classes=None: np.searchsorted(classes, block), dtype=self.classes_.dtype, classes=self.classes_)
```

This avoids placing `self` into the `lambda`, which might cause issues when serializing and sending to other workers.

Reply:
Or, something like the following might work as well and be a bit cleaner actually:

```python
da.map_blocks(np.searchsorted, self.classes_, y)
```

This is probably cheaper to serialize and also keeps the nice searchsorted function name for the diagnostics.

Reply:
```python
return da.map_blocks(np.searchsorted, self.classes_, y, dtype=self.classes_.dtype)
```

Seems to work just fine and is cleaner in my opinion, thanks for the recommendation @mrocklin! Not sure why I used the lambda in the first place...